src/PRASBase/pras_system_size_reducer.jl

```diff
@@ -0,0 +1,91 @@
+#######################################################
+# Surya
+# NREL
+# January 2020
+# Make a smaller PRAS system (with a fewer timesteps)
+# from a bigger PRAS system
+#######################################################
+# Main Function to reduce size of PRAS System
+#######################################################
+function reduce_pras_system_size(pras_system::PRAS.SystemModel,period_of_interest::UnitRange)
+    """
+    reduce_system_size(sys,x)
+
+    Build a smaller PRAS system with 'x' time_steps from a larger PRAS System.
+    
+    ...
+    # Arguments
+    - `sys::SystemModel`: Original PRAS System.
+    - `x::UnitRange`: Period of interest to build smaller PRAS System
+    ...
+
+    # Examples
+    ```julia-repl
+    julia> reduce_system_size(sys,range(100,length=100))
+    SystemModel{100,timestep_length,timeunits,powerunits,energyunits(optional)}
+    ```
+    """
+    symbol = PRAS.unitsymbol(pras_system);
```

Comment:
The unit types could be extracted more cleanly with type parameter matching on the method signature:

```julia
function reduce_pras_system_size(
    pras_system::PRAS.SystemModel{N,L,T,P,E}, period_of_interest::UnitRange
) where {N,L,T,P,E}
```
Then `T`, `P`, `E` can be used directly, to avoid getting the units as strings and then converting them back to types.

Reply:
Type parameter matching is really neat. I didn't know about this. Thanks for letting me know.